hikari/commands.py

```diff
@@ -292,3 +306,133 @@ async def delete(self) -> None:
+        """Set permissions for this command in a specific guild.
+
+        !!! note
+            This overwrites any previously set permissions.
+
+        Parameters
+        ----------
+        guild : hikari.undefined.UndefinedOr[hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]]
+            Object or ID of the guild to set the command permissions in.
+        permissions : typing.Sequence[CommandPermission]
+            Sequence of up to 10 of the permission objects to set.
+
+        Returns
+        -------
+        GuildCommandPermissions
+            Object of the set permissions.
+
+        Raises
+        ------
+        hikari.errors.ForbiddenError
+            If you cannot access the provided application's commands or guild.
+        hikari.errors.NotFoundError
+            If the provided application or command isn't found.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+        return await self.app.rest.set_application_command_permissions(
+            application=self.application_id, command=self.id, guild=guild, permissions=permissions
+        )
+
+
+class CommandPermissionType(int, enums.Enum):
+    """The type of entity a command permission targets."""
+
+    ROLE = 1
+    """A command permission which toggles access for a specific role."""
+
+    USER = 2
+    """A command permission which toggles access for a specific user."""
+
+
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class CommandPermission:
+    """Representation of a permission which enables or disables a command for a user or role."""
+
+    id: snowflakes.Snowflake = attr.field(converter=snowflakes.Snowflake)
+    """Id of the role or user this permission changes the permission's state for."""
+
+    type: typing.Union[CommandPermissionType, int] = attr.field(converter=CommandPermissionType)
+    """The entity this permission overrides the command's state for."""
+
+    is_enabled: bool = attr.field()
+    """Whether this permission marks the relevant command as enabled or disabled."""
```

Comment:
Wouldn't a better name for this be `is_command_enabled`? `is_enabled` sounds like whether the permission is enabled, not the command

Reply:
It's now called has_access